src/main/java/com/lindar/wellrested/vo/WellRestedResponse.java

```diff
@@ -30,6 +30,7 @@ public class WellRestedResponse implements Serializable {
     private @Getter @Setter Map<String, String> responseHeaders;
     private @Getter @Setter boolean             socketTimeout;
     private @Getter @Setter boolean             connectionTimeout;
+    private @Getter @Setter boolean             connectionRequestTimeout;
```

Comment:
Was debating whether or not to add this extra field as it's a bit messy having 3 different fields for timeout flags. In the future this should have but wouldn't be backwards compatible. connectionRequestTimeout could be retried fine as the request was never made so I think it's useful to have.